Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5863,6 +5865,13 @@ void WebPageProxy::didStartProvisionalLoadForFrameShared(Ref<WebProcessProxy>&&
         internals().pageLoadState.didStartProvisionalLoad(transaction, url.string(), unreachableURL.string());
         protectedPageClient->didStartProvisionalLoadForMainFrame();
         closeOverlayedViews();
+        if (auto* websitePolicies = navigation->websitePolicies()) {
```

Comment:
I don't think this is the right place for this logic.  The lambda in WebPageProxy::decidePolicyForNavigationAction receives them from the client and shouldn't put them on the navigation.  It should be retained with a RefPtr and passed to WebPageProxy::receivedNavigationActionPolicyDecision as a separate parameter.  That'll take some untangling that should probably be done in a separate PR.  Then in WebPageProxy::receivedNavigationActionPolicyDecision we should store them on the ProvisionalPageProxy and when committing a ProvisionalPageProxy we should move them from the ProvisionalPageProxy to the WebPageProxy.